ntoskrnl/se/token.c

```diff
@@ -135,34 +135,222 @@ SepDeleteTokenLock(
     ExFreePoolWithTag(Token->TokenLock, TAG_SE_TOKEN_LOCK);
 }
 
-static NTSTATUS
-SepCompareTokens(IN PTOKEN FirstToken,
-                 IN PTOKEN SecondToken,
-                 OUT PBOOLEAN Equal)
+/**
+ * @brief
+ * Compares the privileges and attributes from the given tokens.
+ *
+ * @param[in] FirstToken
+ * The first token.
+ *
+ * @param[in] IndexOfFirstToken
+ * The index of the first token.
+ *
+ * @param[in] SecondToken
+ * The second token.
+ *
+ * @param[in] IndexOfSecondToken
+ * The index of the second token.
+ *
+ * @return
+ * Returns TRUE if both the privileges and the attributes are equal,
+ * FALSE otherwise.
+ */
+static
+BOOLEAN
+SepComparePrivilegesAndAttributesFromTokens(
```

Comment:
Why not inlining this function? It is used only once.

Reply:
I don't see much point in inlining the functions since they're already statically declared as any other private routine of this file, but fine by me I guess. I'll make those functions inlined then.

Reply:
Please do not use inline for static functions which are not part of a header. 

1. Let the compiler be smarter than you on this topic.
2. Let the debugger help you when things go wrong.